src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -274,13 +351,129 @@ internal static byte[][] SegmentedEncodeNamedBitList(byte[] bigEndianBytes, int
                 dataSegment
             };
         }
-        
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of an octet string (byte array).
+        /// </summary>
+        /// <param name="data">The data to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeOctetString(byte[] data)
+        {
+            Debug.Assert(data != null);
+
+            // Because this is not currently public API the data array is not being cloned.
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.OctetString }, 
+                EncodeLength(data.Length),
+                data,
+            };
+        }
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of an object identifier (Oid).
+        /// </summary>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeOid(Oid oid)
+        {
+            if (oid == null)
```

Comment:
The null check here I can probably change to an assert safely; the throws are for things like

```C#
OidCollection ekus = new OidCollection
{
    new Oid("3"), // Doesn't throw
    new Oid("Unresolved Friendly Name"), // Doesn't throw
};
var ekuExt = new X509EnhancedKeyUsageExtension(ekus);
byte[] data = ekuExt.RawData; // throws CryptographicException on Encode
```

It was the only thing that really had the notion of invalid data permeating it, so it's the only one that had to bother to throw.